circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -232,34 +233,72 @@ private class CircuitSymbolProcessor(
                 assistedParams
               )
             FactoryType.UI -> {
+              // State param is optional
               val stateParam =
                 fd.parameters.singleOrNull { parameter ->
                   symbols.circuitUiState.isAssignableFrom(parameter.type.resolve())
                 }
-              if (stateParam == null) {
-                CodeBlock.of(
-                  "%M<%T>·{·%M(%L)·}",
-                  MemberName("com.slack.circuit", "ui"),
-                  CircuitUiState::class.java,
-                  MemberName(packageName, name),
-                  assistedParams
-                )
-              } else {
-                val block =
-                  if (assistedParams.isEmpty()) {
-                    CodeBlock.of("")
-                  } else {
-                    CodeBlock.of(",·%L", assistedParams)
+
+              // Modifier param is required
+              val modifierParam =
+                fd.parameters.singleOrNull { parameter ->
+                  symbols.modifier.isAssignableFrom(parameter.type.resolve())
+                }
+                  ?: run {
+                    logger.error("UI composable functions must have a Modifier parameter!", fd)
+                    return null
                   }
-                CodeBlock.of(
-                  "%M<%T>·{·state·->·%M(%L·=·state%L)·}",
-                  MemberName("com.slack.circuit", "ui"),
-                  stateParam.type.resolve().toTypeName(),
-                  MemberName(packageName, name),
-                  stateParam.name!!.getShortName(),
-                  block
-                )
-              }
+
+              /*
+              Diagram of what goes into generating a function!
+              - State parameter is _optional_ and can be omitted if it's static state.
+                - When omitted, the argument becomes _ and the param is omitted entirely.
+              - <StateType> is either the State or CircuitUiState if no state param is used.
```

Comment:
Thought: I could see someone reading the generated code and getting confused by a param of type `CircuitUiState`. I'm wondering if it might make sense to create  `object NoState : CircuitUiState` and use it here instead.

Reply:
unless we promote NoState to be a public and reusable type, I don't think it makes much of a difference. I have wondered if theres room for a `class StaticState<UiEvent : CircuitUiEvent>(val eventSink: (UiEvent) -> Unit)` that could be useful